src/services/intelligence/preference_storage.py

```
@@ -0,0 +1,545 @@
+        """Look up preferences via the manager's by-type / by-key methods."""
+        with self._lock:
+            if key is not None:
+                row = self._db.get_preference(preference_type.value, key)
+                return [_row_to_preference(row)] if row else []
+            rows = self._db.get_preferences_by_type(preference_type.value)
+            return [_row_to_preference(r) for r in rows]
+
+    def update_preference_confidence(
+        self,
+        preference: Preference,
+        success: bool,
```

Ruff (0.15.11)

[warning] Boolean-typed positional argument in function definition

(FBT001)